File: leetcode-submissions/283_Move_Zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
 
function moveZeroes(nums: number[]): void {
   let z_count=0;
   for(let i=0;i<nums.length;i++){
    if(nums[i]===0){
        z_count++;
        nums.splice(i,1);
    }
   }
   while(z_count--){
    nums.push(0);
   }
};

File: leetcode-submissions/283_Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
     int left=0;
     int right=nums.size()-1;
     while(left<=right){
         if(nums[left]==0 and nums[right]!=0){
             swap(nums[left],nums[right]);
             left++;
             right--;

         }else{
             left++;
             right--;
         }
     } 
     
    }
};